src/prop/prop_proof_manager.h

```diff
@@ -118,6 +118,14 @@ class PropPfManager : protected EnvObj
   /** Return lemmas used in the SAT proof. */
   std::vector<Node> getUnsatCoreLemmas();
 
+  /**
+   * Get inference id for a lemma, e.g. one that appears in the return of
+   * getUnsatCoreLemmas. Note that the inference id will be InferenceId::NONE
+   * if lem is not an unsat core lemma, or if it corresponded e.g. to a lemma
+   * learned via theory propagation.
```

Comment:
We could consider eventually tracking these IDs for propagations as well. Other than the interfaces, in the theory engine/solvers side you think this would make sense?

Reply:
Yes, I think we could associate an inference id with propagations, maybe as part of the `explain` interface.

Reply:
This would have to be done on further PRs.

Reply:
Sure!